File: clients/java1.8_client/client/src/main/java/core/MyElevator.java
package core;

import core.API.Elevator;
import core.API.Passenger;

import java.util.List;

public class MyElevator {
    private Elevator elevator;

    public MyElevator(Elevator elevator) {
        this.elevator = elevator;
    }

    public Integer getId() {
        return elevator.getId();
    }

    public Double getY() {
        return elevator.getY();
    }

    public List<Passenger> getPassengers() {
        return elevator.getPassengers();
    }

    public ElevatorState getState() {
        return ElevatorState.parse(elevator.getState());
    }

    public Double getSpeed() {
        return elevator.getSpeed();
    }

    public Integer getTimeOnFloor() {
        return elevator.getTimeOnFloor();
    }

    public Integer getFloor() {
        return elevator.getFloor();
    }

    public String getType() {
        return elevator.getType();
    }

    public Integer getNextFloor() {
        return elevator.getNextFloor();
    }

    public Elevator getElevator() {
        return elevator;
    }

    public void goToFloor(Integer floor) {
        elevator.goToFloor(floor);
    }
}

File: clients/java1.8_client/client/src/main/java/core/MyElevator.kt
package core

import core.API.Elevator
import core.API.Passenger

class MyElevator(val elevator: Elevator) {
    companion object {
        const val MAX = 20
    }

    val id: Int?
        get() = elevator.id

    val y: Double?
        get() = elevator.y

    val passengers: List<MyPassenger>
        get() = elevator.passengers.map { MyPassenger(it, type == it.type) }

    val currentPassengers: Int
        get() = elevator.passengers.size

    val full: Boolean
        get() = currentPassengers >= MAX

    val empty: Boolean
        get() = currentPassengers == 0

    val state: ElevatorState
        get() = ElevatorState.parse(elevator.state)!!

    val speed: Double?
        get() = elevator.speed

    val timeOnFloor: Int?
        get() = elevator.timeOnFloor

    val floor: Int
        get() = elevator.floor

    val type: String
        get() = elevator.type

    val nextFloor: Int?
        get() = elevator.nextFloor

    fun goToFloor(floor: Int?) {
        elevator.goToFloor(floor)
    }
}
